### ISSUE TITLE
Service Net Admins Email Notifications When User Registers to Service Net

### ISSUE THREAD

**Issue by EnriqueCorado** [benetech/ServiceNet#1473]
Purpose
------
Service Net Admins need a notification to indicate when users have registered on Service Net in order to approve accounts

Users
-----------
Benetech Admins

Features
-----------
Dev please find the best way 
1. Assign roles to benetech users (if not already there) that have the clearance to approve registration accounts 
2. Send email notifications to those users that have access to approve accounts

**Comment by meenupant** [comment]
@johnhbenetech How does the account approval process works for other Benetech products like Bookshare? Can we adopt those best practices here as well?

**Comment by johnhbenetech** [comment]
@meenupant I don't think their process would be relevant. Due to the extreme membership limitations they actually require paperwork to be filled out manually, etc.

**Comment by meenupant** [comment]
@johnhbenetech @EnriqueCorado I researched and there are many ways to implement it. Let's discuss this in a zoom call. It will need to be scoped out in detail including the error notification and redirecting. 

Refer:
http://www.web-site-scripts.com/knowledge-base/article/AA-00747/0/User-Registration-Workflow.html

Please note that each template consists of two parts: one for the subject and one for the body. E.g. "Email confirmation" template has two entries in the Interface Languages:

* 		Email confirmation (subject)
* 		Email confirmation (body)

Keep this in mind when you'll be editing the templates.

The registration workflow is different when the "Moderation Required" option is enabled or disabled.
If the "Moderation Required" option is disabled

1. User submits the registration form.
2. User gets an email with the link for confirmation of their email address ("Email confirmation" template).
3. User gets an email saying that their account has been created ("Successful registration" template).
4. Staff get a notification that the user has been registered ("New user has been registered" template).

If the "Moderation Required" option is enabled

1. User submits the registration form.
2. User gets an email with the link for confirmation of their email address ("Email confirmation" template).
3. User gets an email saying that they need to wait their account approval by Staff ("Wait for activation" template).
4. Staff get an email saying that the user account needs approval ("User approval" template).
5. User gets an email saying that their account has been approved by Staff ("Account approved" template).

CMS reference example:
https://techdocs.broadcom.com/us/en/ca-enterprise-software/layer7-api-management/api-portal-legacy/3-5/manage-the-api-portal/customize-the-api-portal/the-content-management-system/configure-the-approval-processes.html

**Comment by jkondrat** [comment]
@EnriqueCorado @meenupant  what other permissions should Benetech Admin have compared to the existing 'Super Admin'?

**Comment by kkaczmarczyk** [comment]
@EnriqueCorado @meenupant would it be a good idea to add the background image which is used in the Service Net to the email content? We could also stylize the email template to have content in a kinda similar manner as we have cards on SP view.

![image](https://user-images.githubusercontent.com/3207119/106131171-a225af00-6162-11eb-8f33-2d5f0403c646.png)

**Comment by EnriqueCorado** [comment]
@jkondrat give the "Benetech Admin" the same permissions as 'Super Admin' for now while the permissions of 'Benetech Admin' get scoped more 

@kkaczmarczyk please add the background image and style the email template to have the content in a similar way as the record cards in SP view

**Comment by johnhbenetech** [comment]
@jkondrat @EnriqueCorado Can we just not do Benetech Admin right now? Since there is no difference from admin i'd rather wait until we have actual differences to code. This will be a small number of users so it wont be a bug deal to flip them over to a new role if/when we need it

**Comment by EnriqueCorado** [comment]
@jkondrat @johnhbenetech sounds good.

**Comment by johnhbenetech** [comment]
@EnriqueCorado please update description accordingly

**Comment by EnriqueCorado** [comment]
Note: description updated

**Comment by meenupant** [comment]
> @EnriqueCorado @meenupant would it be a good idea to add the background image which is used in the Service Net to the email content? We could also stylize the email template to have content in a kinda similar manner as we have cards on SP view.
> 
> ![image](https://user-images.githubusercontent.com/3207119/106131171-a225af00-6162-11eb-8f33-2d5f0403c646.png)

@kkaczmarczyk Yes, good idea. Please do that. Thanks

**Comment by meenupant** [comment]
We may change the registration fields and the email sent to Service Net Admin should include all the registration fields in the email body except password (if any). I will update the description with this change.

**Comment by EnriqueCorado** [comment]
task updated

**Comment by jkondrat** [comment]
@EnriqueCorado is it okay to include just the basic info like first and last name, email, username and skip the organization stuff? The user doesn't contain that information and in order to include that in the email, we would have to move verification to the core service (where UserProfile exists) which would add to the complexity of this task. II'd say this would 1-2 more story points.

**Comment by EnriqueCorado** [comment]
@jkondrat yeah, just include basic info like first and last name, email, username in the email. the rest we will be able to see in the user profile

**Comment by jkondrat** [comment]
@EnriqueCorado thank you!

**Comment by abaranowska1** [comment]
**Tested with success**
Email notification is sent when the user registers to Service Net. We include just the basic info like first and last name, email, username
![Selection_999(092)](https://user-images.githubusercontent.com/61964176/107775806-16567a00-6d41-11eb-9b8b-40ee09a2f586.png)
![Selection_999(091)](https://user-images.githubusercontent.com/61964176/107775810-1787a700-6d41-11eb-9bc9-df53deb28bef.png)
